yb-voyager/src/cp/yugabyted/yugabyted.go

```diff
@@ -0,0 +1,621 @@
+	cp.createAndSendEvent(&snapshotImportEvent.BaseEvent, "IN PROGRESS", "")
+}
+
+func (cp *YugabyteD) SnapshotImportCompleted(snapshotImportEvent *controlPlane.SnapshotImportCompletedEvent) {
+	cp.createAndSendEvent(&snapshotImportEvent.BaseEvent, "COMPLETED", "")
+}
+
+func (cp *YugabyteD) UpdateImportedRowCount(
+	snapshotImportTableMetrics []*controlPlane.UpdateImportedRowCountEvent) {
+
+	var updateImportedRowCountEvents []*controlPlane.BaseUpdateRowCountEvent
+	for _, updateImportedRowCountEvent := range snapshotImportTableMetrics {
+
+		cp.RWMutex.RLock()
+		lastRowCountUpdateTime, check := cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName]
+		cp.RWMutex.RUnlock()
+
+		if updateImportedRowCountEvent.Status == "COMPLETED" {
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !check {
+			cp.RWMutex.Lock()
+			cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName] = time.Now()
+			cp.RWMutex.Unlock()
+
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !lastRowCountUpdateTime.Add(time.Second * 5).After(time.Now()) {
+			cp.RWMutex.Lock()
+			cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName] = time.Now()
+			cp.RWMutex.Unlock()
+
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		}
+
+	}
+	cp.createAndSendUpdateRowCountEvent(updateImportedRowCountEvents)
+}
+
+func (cp *YugabyteD) MigrationEnded(migrationEndedEvent *controlPlane.MigrationEndedEvent) {
+}
+
+func (cp *YugabyteD) panicHandler() {
+	if r := recover(); r != nil {
+		// Handle the panic for eventPublishers
+		log.Warnf(fmt.Sprintf("Panic occurred:%v", r))
+	}
+}
+
+func (cp *YugabyteD) getConn() (*pgxpool.Pool, error) {
+	var err error
+	err = nil
+	if cp.conn == nil {
+		err = fmt.Errorf("called visualizer_yugabyte_db.get_conn() " +
+			"before visualizer_yugabyte_db.connect()")
+	}
+
+	return cp.conn, err
+}
+
+func (cp *YugabyteD) reconnect() error {
+	var err error
+	cp.disconnect()
+	for attempt := 1; attempt < 5; attempt++ {
+		err = cp.connect()
+		if err == nil {
+			return nil
+		}
+		log.Infof("Failed to reconnect to the target database: %s", err)
+		time.Sleep(time.Second)
+	}
+	return fmt.Errorf("reconnect to target db: %w", err)
+}
+
+func (cp *YugabyteD) connect() error {
+	if cp.conn != nil {
+		return nil
+	}
+	connectionUri := os.Getenv("YUGABYTED_DB_CONN_STRING")
+
+	conn, err := pgxpool.Connect(context.Background(), connectionUri)
+	if err != nil {
+		return fmt.Errorf("connect to target db: %w", err)
+	}
+
+	cp.conn = conn
+	return nil
+}
+
+func (cp *YugabyteD) disconnect() {
+	if cp.conn == nil {
+		return
+	}
+
+	cp.conn.Close()
+	cp.conn = nil
+}
+
+const BATCH_METADATA_TABLE_SCHEMA = "ybvoyager_visualizer"
```

Comment:
The identifier `BATCH_METADATA_TABLE_SCHEMA` has no meaning in the context of the `YugabyteD` class.